elastic/kibana
# [Synthetics] Handle private locations simultaneous edits !!

## Summary

Fixes https://github.com/elastic/kibana/issues/190801 !!

Handle private locations simultaneous edits !! 

Registered a new saved object to handle private locations properly. Instead of using a singleton, now each private location will be represented by it's own saved object.

### Existing private locations
When we are doing any write operation, we migrate them to new kind of saved object and remove the legacy saved object type.


### Testing

- Create multiple private locations on main 
- Switch to branch and create few more locations
- Make sure all private locations are editable, deleteable and have been migrated to new saved object types in the course
